lib/src/common/cache/cache.dart

```diff
@@ -278,11 +351,16 @@ class Cache<TIdentifier, TValue> extends Object with Disposable {
   /// If the [Cache] [isOrWillBeDisposed] then a [StateError] is thrown.
   @mustCallSuper
   Future<Null> remove(TIdentifier id) {
+    _log.finest('remove id: $id');
     _throwWhenDisposed('remove');
+    _isReleased.remove(id);
     if (_cache.containsKey(id)) {
       _cachingStrategy.onWillRemove(id);
       final removedValue = _cache.remove(id);
       return removedValue.then((TValue value) async {
+        if (_applyToItemCallBacks[id] != null) {
+          await Future.wait(_applyToItemCallBacks[id]);
+        }
         await _cachingStrategy.onDidRemove(id, value);
```

Comment:
[line 356](https://github.com/Workiva/w_common/pull/90/files#diff-1fdd0bfc9ee195804c63d431b8409625R356) removes it from the `_isReleased` map synchronously, and then [line 388](https://github.com/Workiva/w_common/pull/90/files#diff-1fdd0bfc9ee195804c63d431b8409625R388) will short circuit the apply.

Reply:
Sounds like this would be good to have a test for.

Reply:
Also, what if a `get` comes in while you're waiting for this to finish? Would it be possible to get an event sequence that looks like:
```
<item not in cache>
item added to cache
didUpdate
apply with callback that takes awhile
item removed from cache
apply callback still hasn't finished
item added to the cache
didUpdate
apply callback finishes
didRemove
<item is still in cache>
```
Could this unexpected sequence of events cause issues for consumers that may not be expecting it?

Reply:
> Sounds like this would be good to have a test for.

[Here's one!](https://github.com/Workiva/w_common/pull/90/files#diff-71e7ef0509871bfd43a03a17b3e52771R450)